internal/service/advancedcluster/resource_advanced_cluster.go

```diff
@@ -765,13 +782,57 @@ func setRootFields(d *schema.ResourceData, cluster *admin.ClusterDescription2024
 		return diag.FromErr(fmt.Errorf(ErrorClusterAdvancedSetting, "global_cluster_self_managed_sharding", clusterName, err))
 	}
 
+	// fields not supported in connV220240530 SDK (mapping to 2023-02-01 API)
+
+	if err := d.Set("replica_set_scaling_strategy", cluster.GetReplicaSetScalingStrategy()); err != nil {
+		return diag.FromErr(fmt.Errorf(ErrorClusterAdvancedSetting, "replica_set_scaling_strategy", clusterName, err))
+	}
+
+	if err := d.Set("redact_client_log_data", cluster.GetRedactClientLogData()); err != nil {
+		return diag.FromErr(fmt.Errorf(ErrorClusterAdvancedSetting, "redact_client_log_data", clusterName, err))
+	}
+
 	if err := d.Set("config_server_type", cluster.GetConfigServerType()); err != nil {
 		return diag.FromErr(fmt.Errorf(ErrorClusterAdvancedSetting, "config_server_type", clusterName, err))
 	}
 
 	if err := d.Set("config_server_management_mode", cluster.GetConfigServerManagementMode()); err != nil {
 		return diag.FromErr(fmt.Errorf(ErrorClusterAdvancedSetting, "config_server_management_mode", clusterName, err))
 	}
+
+	if err := d.Set("pinned_fcv", flattenPinnedFCV(cluster)); err != nil {
+		return diag.FromErr(fmt.Errorf(ErrorClusterAdvancedSetting, "pinned_fcv", clusterName, err))
+	}
+
+	return nil
+}
+
+func warningIfFCVExpiredOrUnpinnedExternally(d *schema.ResourceData, cluster *admin.ClusterDescription20240805) diag.Diagnostics {
```

Comment:
nit: in TPF cluster we normally to pass *diag.Diagnostics as a func param instead of return value so any func can easily add errors and warnings, and you can ask diag.HasError() if needed

Reply:
In SDK v2 I do see we follow the pattern of returning diags up to the root crud functions. Would prefer to stay consistent but do see value in the TPF approach.

Reply:
makes sense, thanks